partiql-planner/src/main/kotlin/org/partiql/planner/internal/transforms/RexConverter.kt

```diff
@@ -120,20 +120,20 @@ internal object RexConverter {
          */
         internal fun visitExprCoerce(node: Expr, ctx: Env, coercion: Rex.Op.Subquery.Coercion = Rex.Op.Subquery.Coercion.SCALAR): Rex {
             val rex = super.visitExpr(node, ctx)
-            return when (isSqlSelect(node)) {
-                true -> {
-                    val select = rex.op as Rex.Op.Select
-                    rex(
-                        CompilerType(PType.typeDynamic()),
-                        rexOpSubquery(
-                            constructor = select.constructor,
-                            rel = select.rel,
-                            coercion = coercion
-                        )
+            // Only need to look into the coercible flag if this is a project value
+            val sfw = node as? Expr.SFW ?: return rex
+            val select = node.select as? Select.Value ?: return rex
+            return if (select.coercible) {
+                val selectRex = rex.op as Rex.Op.Select
+                rex(
+                    CompilerType(PType.typeDynamic()),
+                    rexOpSubquery(
+                        constructor = selectRex.constructor,
+                        rel = selectRex.rel,
+                        coercion = coercion
```

Comment:
When do we coerce SELECT VALUE? I was under the impression we only coerce SQL SELECT

Reply:
I removed this part to reduce noise in this PR.